Library/Homebrew/utils/shell_env.sh

```diff
@@ -0,0 +1,71 @@
+keepkeys=(
+HOMEBREW_*
+GEM_*
+HOME
+PWD
+LANG
+PATH
+)
+
+prune-shell-env() {
+  # Associative arrays are only a Bash 4 feature
+  #declare -A aEnv
+
+  local verbose
+  
+  local envPATH=`which env`
+  local echoPATH=`which echo`
+  local sedPATH=`which sed`
+
+  file=$HOMEBREW_LIBRARY/Homebrew/utils/shell-env.txt
```

Comment:
Quote `$` usage. Also, I'd consider not making this a `txt` file and storing it somewhere in `tmp`.